statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/httpfn/HttpFunctionProvider.java

```diff
@@ -41,12 +48,56 @@ public RequestReplyFunction functionOfType(FunctionType type) {
     if (spec == null) {
       throw new IllegalArgumentException("Unsupported type " + type);
     }
+    return new RequestReplyFunction(
+        spec.states(), spec.maxNumBatchRequests(), buildHttpClient(spec));
+  }
+
+  private RequestReplyClient buildHttpClient(HttpFunctionSpec spec) {
+    // We need to build a UDS HTTP client
+    if (spec.isUnixDomainSocket()) {
+
+      // We need to split the path in order to get the sock file and the path after the sock file
+      Map.Entry<String, String> splittedFilePathAndEndpoint =
+          splitFilePathAndEndpointForUDS(spec.endpoint());
+
+      OkHttpClient specificClient =
+          sharedClient
+              .newBuilder()
+              .socketFactory(
+                  new AFUNIXSocketFactory.FactoryArg(splittedFilePathAndEndpoint.getKey()))
```

Comment:
Unfortunately this does not seems to be working.
I get the following stack trace:
It seems like okhttp is trying to set the socket timeout, before connecting (that makes senes) but the library 
expects first a connection and then setting a timeout (which also makes senes)

```
orker_1           | Caused by: java.net.SocketException: Not open
worker_1           | 	at org.newsclub.net.unix.AFUNIXSocketImpl.validFdOrException(AFUNIXSocketImpl.java:424)
worker_1           | 	at org.newsclub.net.unix.AFUNIXSocketImpl.setOption(AFUNIXSocketImpl.java:504)
worker_1           | 	at org.newsclub.net.unix.AFUNIXSocketImpl$Lenient.setOption(AFUNIXSocketImpl.java:568)
worker_1           | 	at java.net.Socket.setSoTimeout(Socket.java:1159)
worker_1           | 	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:261)
worker_1           | 	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:183)
worker_1           | 	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.java:224)
worker_1           | 	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.java:108)
worker_1           | 	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.java:88)
worker_1           | 	at okhttp3.internal.connection.Transmitter.newExchange(Transmitter.java:169)
worker_1           | 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:41)
worker_1           | 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
worker_1           | 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
worker_1           | 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94)
worker_1           | 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
worker_1           | 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
worker_1           | 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
worker_1           | 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
worker_1           | 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:88)
worker_1           | 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
worker_1           | 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
worker_1           | 	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:229)
worker_1           | 	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:172)
```

Reply:
Ok i'm on it

Reply:
Ok It looks like the socket factory doesn't really work as we expect...

Reply:
In that case I can see two options:
* try out jnrsockets as provided in the original example.
* provide a new RequestReplyClient that is specialized for Unix domain socket, and backed by something else than okhttp.

Reply:
Opened an issue on junixsocket https://github.com/kohlschutter/junixsocket/issues/79

Reply:
Looking at the activity on the two projects, to me it looks a little bit more mantained/mature junixsocket. 
Let's see if the solution with junixsocket is trivial, otherwise I'll switch to jnrsockets.